tests/arch/arm/arm_thread_swap/src/arm_syscalls.c

```diff
@@ -297,12 +297,12 @@ void test_syscall_cpu_scrubs_regs(void)
 	}
 }
 #else
-void test_syscall_cpu_scrubs_regs(void)
+ZTEST_USER(arm_thread_swap, test_syscall_cpu_scrubs_regs)
 {
 	ztest_test_skip();
 }
 
-void test_arm_syscalls(void)
+ZTEST(arm_thread_swap, test_arm_syscalls)
 {
 	ztest_test_skip();
 }
```

Comment:
You'll need to be careful here:

> This file is terrible. Note the `#else` on line 299. This is guarding on `CONFIG_USERSPACE`.

Normally, this is needed since `test_main` _ALWAYS_ adds this function to the test suite, but now we no longer do that nonsense. You can just delete the entire `#else` branch. If you want, you can even update the `CMakeLists.txt` to conditionally include this file using `target_sources_ifdef(CONFIG_USERSPACE src/arm_syscalls.c)` and replace the `GLOB` (your call).

Reply:
Until I'm able to reliably test the builds, I'll make fewer changes here whenever possible.